src/SessionWidget.vala

```diff
@@ -107,6 +140,82 @@ public class A11Y.SessionWidget : Gtk.Grid {
         mouse_settings.bind ("dwell-click-enabled", hover_click, "active", SettingsBindFlags.DEFAULT);
     }
 
+    ~SessionWidget () {
+        if (keyboard_pid != 0) {
+            Posix.kill (keyboard_pid, Posix.Signal.KILL);
+            Posix.waitpid (keyboard_pid, out status, 0);
+            keyboard_pid = 0;
+        }
+
+        if (reader_pid != 0) {
+            Posix.kill (reader_pid, Posix.Signal.KILL);
+            Posix.waitpid (reader_pid, out status, 0);
+            reader_pid = 0;
+        }
+    }
+
+    private void toggle_screen_reader (bool active) {
+        if (active) {
+            try {
+                string[] argv;
+                Shell.parse_argv ("orca --replace", out argv);
+                Process.spawn_async (null, argv, null, SpawnFlags.SEARCH_PATH, null, out reader_pid);
+            } catch (Error e) {
+                warning (e.message);
+            }
+        } else {
+            Posix.kill (reader_pid, Posix.Signal.KILL);
+            Posix.waitpid (reader_pid, out status, 0);
+            reader_pid = 0;
+        }
+    }
+
+    private void toggle_keyboard (bool active) {
+        if (keyboard_window != null) {
+            keyboard_window.visible = active;
+            settings.set_boolean ("greeter", "onscreen-keyboard", active);
+
+            return;
+        }
+
+        int id = 0;
+        int onboard_stdout_fd;
+
+        try {
+            string[] argv;
+            Shell.parse_argv ("onboard --xid", out argv);
+            Process.spawn_async_with_pipes (null, argv, null, SpawnFlags.SEARCH_PATH, null, out keyboard_pid, null, out onboard_stdout_fd, null);
+
+            var f = FileStream.fdopen (onboard_stdout_fd, "r");
+            var stdout_text = new char[1024];
+            f.gets (stdout_text);
+            id = int.parse ((string)stdout_text);
+        } catch (Error e) {
+            warning (e.message);
```

Comment:
Is it right to continue constructing the window if something when wrong with spawning the process?